lib/ansible/module_utils/basic.py

```diff
@@ -1511,15 +1507,42 @@ def _handle_aliases(self, spec=None):
             for alias in aliases:
                 self._legal_inputs.append(alias)
                 aliases_results[alias] = k
-                if alias in self.params:
-                    self.params[k] = self.params[alias]
+                if alias in param:
+                    param[k] = param[alias]
 
         return aliases_results
 
-    def _check_arguments(self, check_invalid_arguments):
+    def _handle_no_log_values(self, spec=None, param=None):
```

Comment:
Above referenced issue is not seen with this patch applied.
eg: Module arg spec
```
argument_spec = dict(
    name=dict(),
    description=dict(no_log=True),
    aggregate=dict(type='list', elements='dict'),
    purge=dict(default=False, type='bool'),
    state=dict(default='present',
               choices=['present', 'absent', 'up', 'down']),
)

aggregate_arg_spec = deepcopy(argument_spec)
del aggregate_arg_spec['aggregate']

argument_spec['aggregate']['options'] = aggregate_arg_spec
```
Module run log
```
"aggregate": [
    {
        "description": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
        "name": "ge-0/0/12",
        "purge": false,
        "state": "present",
    },
    {
        "description": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
        "name": "ge-0/0/13",
        "purge": false,
        "state": "present",
    },
    {
        "description": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
        "name": "ge-0/0/14",
        "purge": false,
        "state": "present",
    }
]
```